homeassistant/components/tailwind/coordinator.py

```diff
@@ -0,0 +1,40 @@
+"""Data update coordinator for Tailwind."""
+from datetime import timedelta
+
+from gotailwind import Tailwind, TailwindDeviceStatus, TailwindError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER
+
+
+class TailwindDataUpdateCoordinator(DataUpdateCoordinator[TailwindDeviceStatus]):
+    """Class to manage fetching Tailwind data."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the coordinator."""
+        self.config_entry = entry
+        self.tailwind = Tailwind(
+            host=entry.data[CONF_HOST],
+            token=entry.data[CONF_TOKEN],
+            session=async_get_clientsession(hass),
+        )
+        super().__init__(
+            hass,
+            LOGGER,
+            name=f"{DOMAIN}_{entry.data[CONF_HOST]}",
+            update_interval=timedelta(seconds=5),
+        )
```

Comment:
```suggestion
    def __init__(self, hass: HomeAssistant) -> None:
        """Initialize the coordinator."""
        super().__init__(
            hass,
            LOGGER,
            name=f"{DOMAIN}_{entry.data[CONF_HOST]}",
            update_interval=timedelta(seconds=5),
        )
        self.tailwind = Tailwind(
            host=self.config_entry.data[CONF_HOST],
            token=self.config_entry.data[CONF_TOKEN],
            session=async_get_clientsession(hass),
        )
```

Reply:
Oh lol, I now realise you reference the entry data in the name, hmm

Reply:
Sorry I don't prefer it.

The client is used by the Super, and should be set up before that IMHO.

Reply:
Hmm I guess this could work...

Reply:
Ok, no indeed, the reference in the coordinators name is making this not work. I can remove the assignment though.

Reply:
Remove the unneeded assignment here: [b75bbac](https://github.com/home-assistant/core/pull/105926/commits/b75bbac6ac49f87d12732db780d363e0c24cbedd)